.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,130 @@
+name: Node Agent CI
+
+on: [push, pull_request]
```

Comment:
I'm noticing the branch + PR plus matrix jobs gets pretty noisy. Not a big deal but we can consider limiting to certain branches in the future.